marahgh/assignment3/wordSearch.cpp

```diff
@@ -1,174 +1,17 @@
-  ~Grid() {
-    if (matrix == nullptr) {
-      return;
-    }
-    for (int i = 0; i < rows; i++) {
-      delete[] matrix[i];
-    }
-    delete[] matrix;
-  }
-
-  /*returns the grid's rows number*/
-  int getRows() { return rows; }
-
-  /*returns the grid's columns number*/
-  int getCols() { return cols; }
-
-  /*return true if the grid is empty, and false otherwise*/
-  bool isEmpty() { return empty; }
-  /*returns the i'th row in the grid if the grid is not empty,and nullptr
-   * otherwise*/
-  char* operator[](int i) {
-    if (isEmpty()) {
-      return nullptr;
-    }
-    return matrix[i];
-  }
-};
-
-/*Dictionary class.
-Dictionary words are lowercase alphabet words*/
-class Dictionary {
-  vector<string> dictionary;
-
-  /****************************************************************************/
-  /* a private auxiliary method of isPrefix,
-  checks whether w2 is a prefix of w1*/
-  bool isWordPrefix(string w1, string w2) {
-    if (w1.size() < w2.size()) {
-      return false;
-    }
-    for (int i = 0; i < w2.size(); i++) {
-      if (w1[i] != w2[i]) {
-        return false;
-      }
-    }
-    return true;
-  }
-
-  /*Removes hyphens("-") and converts letters to lower case(returns a legal
-  dictionary word)
-  if the word contains alphabet letters and hyphens, other wise returns an empty
-  word */
-  string fixWord(string word) {
-    string fixed = "";
-    /*a loop to remove hyphens*/
-    for (char c : word) {
-      if (c != '-') {
-        fixed.append(1, tolower(c));
-      }
-    }
-    /*checks if the fixed word is dictionary legal word
-    (contains only lowercase alphabet letters)*/
-    for (char c : fixed) {
-      if (c < 'a' || c > 'z') {
-        return "";
-      }
-    }
-    return fixed;
-  }
-  /****************************************************************************/
- public:
-  /*creates an empty dictionary*/
-  Dictionary() { dictionary = vector<string>(); }
-
-  /*Returns whether the given string is a valid word*/
-  bool isWord(string word) {
-    return isFoundInVector(dictionary, fixWord(word));
-  }
-
-  /*Adds a new non- empty word to the dictionary*/
-  bool addWord(string word) {
-    string fixed = fixWord(word);
-    if (isWord(fixed) || fixed.empty()) {
-      return false;
-    }
-    dictionary.push_back(fixed);
-    return true;
-  }
-
-  /*Returns whether the given string is a prefix of at least one word in the
-dictionary*/
-  bool isPrefix(string word) {
-    for (string w : dictionary) {
-      if (isWordPrefix(w, word)) {
-        return true;
-      }
-    }
-    return false;
-  }
-};
-
 /*An auxiliary function of find word, receives a grid,
 cell cordinates to start searchnig a word from, a dictionary.
 returns the legal words starts form cell i,j int the words vector*/
-void findWord(Grid& grid, int i, int j, Dictionary dictionary, string word,
-              vector<string>& words) {
+void findWord(Grid& grid, Grid& visited, int i, int j, Dictionary dictionary,

```

Comment:
Reusing the Grid class for visited it's a bit of a stretch :)
Nevertheless, you could use 0 & 1 (instead of '0' & '1') to make the code a bit more readable, since you can do things like `if (visited[i][j]) { ... } ` (you could even use true and false, but that is also a stretch :) )